amazon-kinesis-client/src/main/java/software/amazon/kinesis/leases/LeaseRefresher.java

```diff
@@ -243,4 +243,18 @@ default void updateLeaseWithMetaInfo(Lease lease, UpdateField updateField)
      */
     ExtendedSequenceNumber getCheckpoint(String leaseKey)
             throws ProvisionedThroughputException, InvalidStateException, DependencyException;
+
+    /**
+     * Replaces old lease record with a new lease record.
+     * @param oldLease Old lease which is to be deleted
+     * @param newLease New lease which is to be created
+     * @return
+     * @throws DependencyException
+     * @throws InvalidStateException
+     * @throws ProvisionedThroughputException
+     */
+    default boolean replaceLease(Lease oldLease, Lease newLease)
```

Comment:
the concern with this approach is LeaseRefresher is not marked as InternalKinesisApi and if some customers are implementing it or overriding the DynamoDBLeaseRefresher then the contract can break

Reply:
It will not break since there is a default implementation. Or do you have some other concern around compatibility?

Reply:
an UOE is not a good default implementation. Maybe the reason it is backward compatible is it is invoked only when performing the upgrade from single -> multistream mode and in those cases they need to provide a valid implementation to get it to work. That sounds reasonable to me then.